### ISSUE TITLE
OrangePi R1 Plus LTS integration

### ISSUE THREAD

**Issue by DaveWK** [Marcin666/RK3328_Uboot_SPI#7]
Hi,
This looks really well packaged for my use case -- I have an OrangePi R1 Plus LTS which is very similar to the nanopi r2s. I do not use Ubuntu/Debian, and a lot of the builders out there are out of date and/or force building the entire image and packages even if you just want to leverage the u-boot build for non-debian purposes. Even worse, the builders include a bunch of debian-specific bootstrapping for the build host. so thanks for organizing this so it's easy for someone to hack at u-boot without becoming a debian maximalist and building a bunch of unnecessary packages.

Recently there was a pull request put up for u-boot for he orange pi r1 plus lts to add it to u-boot but it has not landed yet:

http://patchwork.ozlabs.org/project/uboot/list/?series=351783

I was wondering about the version of the dtb you're using here and the version of u-boot. I has some trouble getting the 2 patches in that series to apply to the u-boot mainline branch and was wondering what version you may have forked off at, as the opi-r1+lts  inherits a lot of it's config from the nanopi-r2s variant

**Comment by MarcinWad** [comment]
What dtb file? This one for FIT image?

**Comment by dmikushin** [comment]
Hi @DaveWK , the U-boot used here is actually mainline, it's not a fork. But the changes @MarcinWad is applying to it for the SPI support could of course invalidate the patches that you have for the original code. You did not mention whether you need SPI support in your case. If not, skip the SPI support changes and apply your patches. Please show us here the log of what you do and how it did not work.

**Comment by DaveWK** [comment]
Hi,
So my idea was to load the SPI on my OrangePI r1 LTS with a u-boot image that will always just load whatever the EFI partition is on MMC, so that whenever I put a new linux image on the MMC I wouldn't have to flash the u-boot sectors to the MMC at all. Does that make sense?

**Comment by dmikushin** [comment]
Yes, so you want to preserve a fixed bootloader with the system and use it always, ignoring the bootloaders that are possibly shipped with the official SD cards of different distros.
